monai/metrics/panoptic_quality.py

```diff
@@ -0,0 +1,224 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union
+
+import torch
+
+from monai.metrics.utils import do_metric_reduction, remap_instance_id
+from monai.utils import MetricReduction, optional_import
+
+from .metric import CumulativeIterationMetric
+
+linear_sum_assignment, _ = optional_import("scipy.optimize", name="linear_sum_assignment")
+
+__all__ = ["PanopticQualityMetric", "compute_panoptic_quality"]
+
+
+class PanopticQualityMetric(CumulativeIterationMetric):
+    """
+    Compute panoptic quality between two tensors.
+    Input `y_pred` is compared with ground truth `y`.
+    `y_preds` and `y` should only have non-negative integer values, where each distinct value correspond to a
+    distinct instance.
+    `y_preds` and `y` can be a list of channel-first Tensor (CHW[D]) or a batch-first Tensor (BCHW[D]) and the
+    number of channel should be 1.
+
+    This class refers to the implementation in:
+    https://github.com/TissueImageAnalytics/CoNIC/blob/main/misc/utils.py
+    This class also refers to the following paper:
+    https://openaccess.thecvf.com/content_CVPR_2019/papers/Kirillov_Panoptic_Segmentation_CVPR_2019_paper.pdf
+
+    Example of the typical execution steps of this metric class follows :py:class:`monai.metrics.metric.Cumulative`.
+
+    Args:
+        reduction: define mode of reduction to the metrics,
+            available reduction modes: {``"none"``, ``"mean"``, ``"sum"``, ``"mean_batch"``, ``"sum_batch"``},
+            default to ``"mean"``. if "none", will not do reduction.
+        remap: whether to remap `y_pred` and `y` to ensure contiguous ordering of instance id.
+        match_iou: IOU threshould to determine the paring between `y_pred` and `y`. Usually,
+            it should >= 0.5, the paring between instances of `y_pred` and `y` are identical.
+            If set `match_iou` < 0.5, this function uses Munkres assignment to find the
+            maximal amout of unique paring.
+        smooth_nr: a small constant added to the numerator to avoid zero.
+
+    """
+
+    def __init__(
+        self,
+        reduction: Union[MetricReduction, str] = MetricReduction.MEAN,
+        remap: bool = True,
+        match_iou: float = 0.5,
+        smooth_nr: float = 1e-6,
+    ) -> None:
+        super().__init__()
+        self.reduction = reduction
+        self.remap = remap
+        self.match_iou = match_iou
+        self.smooth_nr = smooth_nr
+
+    def _compute_tensor(self, y_pred: torch.Tensor, y: torch.Tensor):  # type: ignore
+        """
+        Args:
+            y_pred: input data to compute, typical panoptic segmentation model output.
+                It must be in the form of B1HW[D] and has integer type.
+            y: ground truth. It must has the same shape as `y_pred` and has integer type.
```

Comment:
"It must be in the form of B1HW[D] and **have** integer type.
y: ground truth. It must **have** the same shape as `y_pred` and **have** integer type."